src/main/java/au/com/dmg/fusion/response/paymentresponse/PaymentResponse.java

```diff
@@ -196,6 +197,39 @@ public PaymentResponse build() {
                         + "Please set the value by \"poiData()\". "
                         + "The properties \"response\", \"saleData\" and \"poiData\" are required.");
             }
+            //PartialPayment Logic
+            if (this.response.getResult() == ResponseResult.Success
+            && !(this.paymentResult == null)
+            && !(this.paymentResult.getAmountsResp() == null)){
+                AmountsResp amountsResp = this.paymentResult.getAmountsResp();
+
+                // Check for nulls, there's logic for some amounts being null, we don't want to change that
+                BigDecimal surcharge = Optional.ofNullable(amountsResp.getSurchargeAmount())
+                        .orElse(BigDecimal.ZERO);
+                BigDecimal tip = Optional.ofNullable(amountsResp.getTipAmount())
+                        .orElse(BigDecimal.ZERO);
+                BigDecimal cashback = Optional.ofNullable(amountsResp.getCashBackAmount())
+                        .orElse(BigDecimal.ZERO);
+                BigDecimal reqAmt = Optional.ofNullable(amountsResp.getRequestedAmount())
+                        .orElse(BigDecimal.ZERO);
+                BigDecimal authPartialAmt = Optional.ofNullable(amountsResp.getPartialAuthorizedAmount())
+                        .orElse(BigDecimal.ZERO);
+
+                if((surcharge.signum() == 0)
```

Comment:
Just a minor one - can you please use intValue() instead of signum() in all cases where signum() is used? Thanks.